mono/metadata/handle.h

```diff
@@ -307,24 +311,6 @@ void mono_handle_verify (MonoRawHandle handle);
 #define HANDLE_OWNER_STRINGIFY(file,lineno) (const char*) (file ":" STRINGIFY(lineno))
 #endif
 
-
```

Comment:
Why does this needs to be moved?

Reply:
At some point I could not compile w/o moving it. Not necessarily this PR but a much larger one. I believe MonoExceptionHandle was not available in exception-internals.h maybe? I would have to try again without the change. If it works, or this PR works, then it can be restored for now or definitely. Let me try.

Reply:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
../../doltlibtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../.. -I../../mono -I../../libgc/include -I../../mono/eglib -I../../mono/eglib -fvisibility=hidden -D_THREAD_SAFE -DGC_MACOSX_THREADS -DUSE_MMAP -DUSE_MUNMAP -g -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes  -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-cast-qual -Wwrite-strings -Wno-switch -Wno-switch-enum -Wno-unused-value -Wno-attributes -Wno-format-zero-length -Qunused-arguments -Wno-unused-function -Wno-tautological-compare -Wno-parentheses-equality -Wno-self-assign -Wno-return-stack-address -Wno-constant-logical-operand -Wno-zero-length-array   -ggdb -O0 -std=gnu99 -fno-strict-aliasing -fwrapv -DMONO_DLL_EXPORT -g -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes  -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-cast-qual -Wwrite-strings -Wno-switch -Wno-switch-enum -Wno-unused-value -Wno-attributes -Wno-format-zero-length -Qunused-arguments -Wno-unused-function -Wno-tautological-compare -Wno-parentheses-equality -Wno-self-assign -Wno-return-stack-address -Wno-constant-logical-operand -Wno-zero-length-array -Werror-implicit-function-declaration -MT libmonoutils_la-mono-error.lo -MD -MP -MF .deps/libmonoutils_la-mono-error.Tpo -c -o libmonoutils_la-mono-error.lo `test -f 'mono-error.c' || echo './'`mono-error.c
In file included from mono-error.c:14:
./mono-error-internals.h:227:1: error: unknown type name 'MonoExceptionHandle'
MonoExceptionHandle
^
./mono-error-internals.h:230:1: error: unknown type name 'MonoExceptionHandle'
MonoExceptionHandle
^
2 errors generated.
make[5]: *** [libmonoutils_la-mono-error.lo] Error 1
```

So I added #include handle.h

```
zero-length-array -Werror-implicit-function-declaration -MT libmonoutils_la-mono-threads-posix.lo -MD -MP -MF .deps/libmonoutils_la-mono-threads-posix.Tpo -c -o libmonoutils_la-mono-threads-posix.lo `test -f 'mono-threads-posix.c' || echo './'`mono-threads-posix.c
In file included from mono-threads-posix.c:20:
In file included from ../../mono/metadata/gc-internals.h:17:
In file included from ../../mono/metadata/object-internals.h:14:
In file included from ../../mono/metadata/threads-types.h:19:
In file included from ../../mono/metadata/handle.h:23:
../../mono/utils/mono-error-internals.h:228:1: error: unknown type name 'MonoExceptionHandle'
MonoExceptionHandle
^
../../mono/utils/mono-error-internals.h:231:1: error: unknown type name 'MonoExceptionHandle'
MonoExceptionHandle
^
2 errors generated.
```

It looks maybe circular, I didn't look long today.